src/intents.cs

```diff
@@ -14998,4 +15087,102 @@ interface INHangUpCallIntentResponse {
+		NativeHandle Constructor (INUnsendMessagesIntentResponseCode code, [NullAllowed] NSUserActivity userActivity);
+
+		[Export ("code")]
+		INUnsendMessagesIntentResponseCode Code { get; }
+	}
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (INIntent))]
+	interface INUnsendMessagesIntent {
+		[Export ("initWithMessageIdentifiers:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string [] messageIdentifiers);
+
+		[NullAllowed, Export ("messageIdentifiers", ArgumentSemantic.Copy)]
+		string [] MessageIdentifiers { get; }
+	}
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	interface INMessageLinkMetadata : NSCopying, NSSecureCoding {
+		[Export ("initWithSiteName:summary:title:openGraphType:linkURL:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string siteName, [NullAllowed] string summary, [NullAllowed] string title, [NullAllowed] string openGraphType, [NullAllowed] NSUrl linkUrl);
+
+		[NullAllowed, Export ("siteName")]
+		string SiteName { get; set; }
+
+		[NullAllowed, Export ("summary")]
+		string Summary { get; set; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; set; }
+
+		[NullAllowed, Export ("openGraphType")]
+		string OpenGraphType { get; set; }
+
+		[NullAllowed, Export ("linkURL", ArgumentSemantic.Copy)]
+		NSUrl LinkUrl { get; set; }
+	}
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (INIntentResponse))]
+	[DisableDefaultCtor]
+	interface INEditMessageIntentResponse {
+		[Export ("initWithCode:userActivity:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (INEditMessageIntentResponseCode code, [NullAllowed] NSUserActivity userActivity);
+
+		[Export ("code")]
+		INEditMessageIntentResponseCode Code { get; }
+	}
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (INIntent))]
+	interface INEditMessageIntent {
+		[Export ("initWithMessageIdentifier:editedContent:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string messageIdentifier, [NullAllowed] string editedContent);
+
+		[NullAllowed, Export ("messageIdentifier")]
+		string MessageIdentifier { get; }
+
+		[NullAllowed, Export ("editedContent")]
+		string EditedContent { get; }
+	}
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
+	interface INEditMessageIntentHandling {
+		[Abstract]
+		[Export ("handleEditMessage:completion:")]
+		void HandleEditMessage (INEditMessageIntent intent, Action<INEditMessageIntentResponse> completion);
+
+		[Export ("confirmEditMessage:completion:")]
+		void ConfirmEditMessage (INEditMessageIntent intent, Action<INEditMessageIntentResponse> completion);
+
+		[Export ("resolveEditedContentForEditMessage:withCompletion:")]
+		void ResolveEditedContent (INEditMessageIntent intent, Action<INStringResolutionResult> completion);
```

Comment:
Should these be async?

Reply:
Ah I merged and your comments popped up. Async can't be applied in protocols